### ISSUE TITLE
Key size too big for index - 

### ISSUE THREAD

**Issue by pjeh** [eXpandFramework/eXpand#465]
When running a web or win application 

**To Reproduce**
After adding Expand (Excel Importer) to an existing project and running the application against a Firebird database server the following error occurs:

Exception:
Unable to create 'Index' 'iTypeName_SequenceObject'. Parent: 'SequenceObject'. Error: Executing Sql 'create unique index \"iTypeName_SequenceObject\" on \"SequenceObject\"(\"TypeName\")' with parameters '' exception 'FirebirdSql.Data.FirebirdClient.FbException (0x80004005): unsuccessful metadata update\r\nkey size too big for index iTypeName_SequenceObject ---> FirebirdSql.Data.Common.IscException: Exception of type 'FirebirdSql.Data.Common.IscException' was thrown.\r\n   at FirebirdSql.Data.Client.Gds.GdsConnection.ReadStatusVector()\r\n   at FirebirdSql.Data.Client.Gds.GdsConnection.ReadResponse()\r\n   at FirebirdSql.Data.Client.Gds.GdsDatabase.ReadResponse()\r\n   at FirebirdSql.Data.Client.Gds.GdsStatement.Execute()\r\n   at FirebirdSql.Data.FirebirdClient.FbCommand.ExecuteCommand(CommandBehavior behavior, Boolean returnsSet)\r\n   at FirebirdSql.Data.FirebirdClient.FbCommand.ExecuteNonQuery()\r\n   at FirebirdSql.Data.FirebirdClient.FbCommand.ExecuteNonQuery()\r\n   at DevExpress.Xpo.DB.ConnectionProviderSql.<>c__DisplayClass109_0.<InternalExecSql>b__0()\r\n   at DevExpress.Xpo.Logger.LogManager.Log[T](String category, LogHandler`1 handler, MessageHandler`1 createMessageHandler)\r\n   at DevExpress.Xpo.DB.ConnectionProviderSql.InternalExecSql(IDbCommand command)\r\n   at DevExpress.Xpo.DB.ConnectionProviderSql.ExecSql(Query query)'"	DevExpress.Xpo.DB.Exceptions.UnableToCreateDBObjectException

Inner Exception:
"Executing Sql 'create unique index \"iTypeName_SequenceObject\" on \"SequenceObject\"(\"TypeName\")' with parameters '' exception 'FirebirdSql.Data.FirebirdClient.FbException (0x80004005): unsuccessful metadata update\r\nkey size too big for index iTypeName_SequenceObject ---> FirebirdSql.Data.Common.IscException: Exception of type 'FirebirdSql.Data.Common.IscException' was thrown.\r\n   at FirebirdSql.Data.Client.Gds.GdsConnection.ReadStatusVector()\r\n   at FirebirdSql.Data.Client.Gds.GdsConnection.ReadResponse()\r\n   at FirebirdSql.Data.Client.Gds.GdsDatabase.ReadResponse()\r\n   at FirebirdSql.Data.Client.Gds.GdsStatement.Execute()\r\n   at FirebirdSql.Data.FirebirdClient.FbCommand.ExecuteCommand(CommandBehavior behavior, Boolean returnsSet)\r\n   at FirebirdSql.Data.FirebirdClient.FbCommand.ExecuteNonQuery()\r\n   at FirebirdSql.Data.FirebirdClient.FbCommand.ExecuteNonQuery()\r\n   at DevExpress.Xpo.DB.ConnectionProviderSql.<>c__DisplayClass109_0.<InternalExecSql>b__0()\r\n   at DevExpress.Xpo.Logger.LogManager.Log[T](String category, LogHandler`1 handler, MessageHandler`1 createMessageHandler)\r\n   at DevExpress.Xpo.DB.ConnectionProviderSql.InternalExecSql(IDbCommand command)\r\n   at DevExpress.Xpo.DB.ConnectionProviderSql.ExecSql(Query query)'"}	System.Exception {DevExpress.Xpo.DB.Exceptions.SqlExecutionErrorException

This occurs whether the database file exists or is being created during the database update.  

It also occurs when creating an Expand solution from scratch, not just adding to an existing solution

**Comment by apobekiaris** [comment]
hi this snippet can fix the problem.

```cs
public override void CustomizeTypesInfo(ITypesInfo typesInfo) {
            base.CustomizeTypesInfo(typesInfo);
            var member = (XafMemberInfo)( typesInfo.FindTypeInfo(typeof(SequenceObject))).FindMember(nameof(SequenceObject.TypeName));
            member.RemoveAttribute(new SizeAttribute(1024));
            member.AddAttribute(new SizeAttribute(FireBirdIndexLImit));
        }
```

**Comment by pjeh** [comment]
It makes no difference what size I set for member.AddAttribute(new SizeAttribute(FireBirdIndexLImit));
However, it does when I comment out that line.

**Comment by pjeh** [comment]
Is there a reason why I should apply an attribute size to this member?

**Comment by apobekiaris** [comment]
>Unable to create 'Index' 'iTypeName_SequenceObje

the default index cannot be created cause the TypeName size is invalid in your case, so I suggested to override it. However if the db already exists overriding makes no difference. Solution to this (add the code I suggested so it won't create invalid config again) and drop the SequenceObject table and let it recreated it or just simply change the member size manually and since the memmber exist it won't be modified as per declaration.
